steam/ext/dota2/state.py

```diff
@@ -0,0 +1,29 @@
+"""Licensed under The MIT License (MIT) - Copyright (c) 2020-present James H-B. See LICENSE"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from ..._gc import GCState as GCState_
+from ...app import DOTA2
+from .protobufs.client_messages import CMsgClientToGCGetProfileCard
+from .protobufs.common import CMsgDOTAProfileCard
+from .protobufs.sdk import CMsgClientHello
+
+if TYPE_CHECKING:
+    from .client import Client
+
+
+class GCState(GCState_[Any]):  # todo: implement basket-analogy for dota2
```

Comment:
This needs to have support for _store_user overwriting the type of the User stored as you have a custom one